File: apps/catalog-ui/react/src/app/components/catalog/CatalogConstants.ts
import { CatalogOptions } from "catalog-shared";

export const DIR_OPTIONS = CatalogOptions.getInstance().dirOptions.map((o: any) => ({ id: o.id, label: o.label }));
export const SIZE_OPTIONS = CatalogOptions.getInstance().sizeOptions.map((o: any) => ({ id: o.id.toLowerCase(), label: o.label }));
export const RADIUS_OPTIONS = CatalogOptions.getInstance().radiusOptions.map((o: any) => ({ id: o.id, label: o.label }));
export const RADIUS_FRACTION_MAP: Record<string, number> = Object.fromEntries(
	CatalogOptions.getInstance().radiusOptions.map((o: any) => [o.id, CatalogOptions.getInstance().radiusFraction(o.id)]),
);
export const MAX_CONTAINER_RADIUS_MAP: Record<string, number> = Object.fromEntries(
	CatalogOptions.getInstance().radiusOptions.map((o: any) => [o.id, CatalogOptions.getInstance().maxContainerRadius(o.id)]),
);
export const PANEL_VARIANT_OPTIONS = CatalogOptions.getInstance().panelVariantOptions.map((o: any) => ({ id: o.id, label: o.label }));
export const PANEL_SIDE_OPTIONS = CatalogOptions.getInstance().panelSideOptions.map((o: any) => ({ id: o.id, label: o.label }));

File: apps/catalog-ui/compose/src/commonMain/kotlin/CatalogConstants.kt
import catalog.CatalogOptions
import com.uikit.components.composites.button.ButtonSize
import com.uikit.foundation.ComponentSize

internal val SIZE_OPTIONS = CatalogOptions.sizeOptions.map { it.id to it.label }

internal val RADIUS_OPTIONS = CatalogOptions.radiusOptions.map { it.id to it.label }

internal val RADIUS_FRACTION_MAP = CatalogOptions.radiusOptions.associate { it.id to CatalogOptions.radiusFraction(it.id) }

internal val MAX_CONTAINER_RADIUS_MAP = CatalogOptions.radiusOptions.associate { it.id to CatalogOptions.maxContainerRadius(it.id) }

internal fun sizeFromId(id: String): ComponentSize =
	ComponentSize.entries.first { it.name == id }

internal fun buttonSizeFromComponentSize(cs: ComponentSize): ButtonSize =
	ButtonSize.entries.first { it.name == cs.name }

internal val PANEL_VARIANT_OPTIONS = CatalogOptions.panelVariantOptions.map { it.id to it.label }

internal val PANEL_SIDE_OPTIONS = CatalogOptions.panelSideOptions.map { it.id to it.label }
